core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java

```diff
@@ -419,8 +420,25 @@ public void setImageUrl(String imageUrl) {
         return items;
     }
 
+    private List<FeedItem> dedupItems(List<FeedItem> list) {
+        if (list == null) {
+            return null;
+        }
+        ArrayList<String> seen = new ArrayList<>();
+        Iterator<FeedItem> it = list.iterator();
+        while (it.hasNext()) {
+            FeedItem item = it.next();
+            if (seen.indexOf(item.getItemIdentifier()) == -1) {
+                seen.add(item.getItemIdentifier());
+            } else {
+                it.remove();
+            }
+        }
+        return list;
+    }
+
     public void setItems(List<FeedItem> list) {
-        this.items = list;
+        this.items = dedupItems(list);
```

Comment:
@ByteHamster doing the deduplication when `setIems` should guarantee there won't be dups

Reply:
This would then be done every single time that we load items from the database. I think it should rather be done in `DBTasks.updateFeed`. Maybe the loop there could also be simplified a bit (it looks unnecessarily complicated to me but maybe I am just overlooking something)

Reply:
I don't think I can do the dedup in `DBTasks.updateFeed` for example

1) I enter a RSS feed with duplicate in Subscribe by RSS
2) When the popup is shown, I would need to dedup in the preview This is before `DBTasks.updateFeed()`

I do see your point about doing the dedup way too often.  let me think about how to eliminate deduplication (maybe via an option to `setItem(items, boolDedup)` and we only set `boolDedup` on new feeds?

Reply:
@ByteHamster I finally understand what you are saying, `setItems()` are called during database loading.  I have fixed what you wanted, calling deduplication only during parsing

Reply:
Thanks, that looks more clean to me :)